OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -263,6 +263,14 @@ public virtual Barrel CheckFire(Actor self, IFacing facing, Target target)
 				FireDelay = Util.ApplyPercentageModifiers(Weapon.ReloadDelay, modifiers);
 				Burst = Weapon.Burst;
 
+				if (Weapon.ReloadSound != null)
```

Comment:
The report case above checks for both null and the list being nonempty. You should probably do the same here, or prove that the above code is being excessively paranoid.